# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/base/PresShell.cpp
# Commit: f6ae14913216
# Full Hash: f6ae1491321623380e67115dc649d6b22ff84c6d
# Author: Hiroyuki Ikezoe <hikezoe@mozilla.com>
# Date: 2019-05-15 15:58:16
# Description:
#   Bug 1551241 - Add a check for the case where the embedder element is nullptr. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D30982
# ==============================================================================

diff -r f613e818e435 -r f6ae14913216 layout/base/PresShell.cpp
--- a/layout/base/PresShell.cpp	Wed May 15 10:45:25 2019 +0000
+++ b/layout/base/PresShell.cpp	Wed May 15 10:55:34 2019 +0000
@@ -10921,7 +10921,15 @@
     // Propagate to children.
     for (BrowsingContext* child : bc->GetChildren()) {
       Element* embedderElement = child->GetEmbedderElement();
-      MOZ_ASSERT(embedderElement);
+      if (!embedderElement) {
+        // TODO: We shouldn't need to null check here since `child` and the
+        // element returned by `child->GetEmbedderElement()` are in our
+        // process (the actual browsing context represented by `child` may not
+        // be, but that doesn't matter).  However, there are currently a very
+        // small number of crashes due to `embedderElement` being null, somehow
+        // - see bug 1551241.  For now we wallpaper the crash.
+        continue;
+      }
 
       bool embedderFrameIsHidden = true;
       if (auto embedderFrame = embedderElement->GetPrimaryFrame()) {
